### PR TITLE
Fix erfinv float32 rounding

### PR DESCRIPTION
**Pull request by sanjays2007**

### CODE REVIEW

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
Thanks for submitting this PR! This is a very helpful fix to resolve accuracy and reproducibility drift between TensorFlow's eager execution and XLA-compiled paths.
  
  Before we can merge this, there are a few issues that need to be addressed:
  
  1. **Compilation Failures**:
     - There is a syntax error in `tensorflow/core/kernels/cwise_ops.h`: the closing brace `};` was added at the end of the `operator()` definition, closing the `scalar_erfinv_op` struct early and breaking the build.
     - `Eigen::numext::nextafter` is only defined in Eigen for `bfloat16`. Calling it for standard `float` and `double` instantiations does not compile. Please use `std::nextafter(T(1), T(0))` instead.
  
  2. **Implement the fix in the Vector Path (`packetOp`)**:
     Updating `operator()` only fixes the scalar path, meaning vectorized evaluations will still suffer from the rounding issues. Please apply standard SIMD/packet checks using Eigen's packet math helper. You can write it like this:
     ```cpp
     template <typename Packet>
     EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Packet packetOp(const Packet& x) const {
       Packet half = pset1<Packet>(T(0.5));
       Packet p = pmadd(half, x, half);
       Packet one = pset1<Packet>(T(1.0));
       Packet boundary = pset1<Packet>(std::nextafter(T(1.0), T(0.0)));
       Packet mask = pandnot(pcmp_lt(x, one), pcmp_lt(p, one));
       p = pselect(mask, boundary, p);
       Packet y = pndtri<Packet>(p);
       Packet half_sqrt = pset1<Packet>(T(M_SQRT1_2));
       return pmul(y, half_sqrt);
     }
  
3.  Separate Unrelated Changes: This PR currently includes modifications to tensorflow/python/keras/layers/convolutional_transpose_validation_test.py and exception checks in nn_ops.py which are unrelated. Please rebase or reset your branch to exclude these and focus purely on the erfinv correctness change.

4. Add Regression Tests: Please add unit tests targeting the boundary inputs (e.g., passing values like np.nextafter(np.float32(1.0), np.float32(0.0))) to ensure erfinv remains finite and behaves symmetrically near boundaries.
